trigger.js

```diff
@@ -206,6 +215,44 @@ class TriggerHappy {
         Hooks.once('updateToken', () => this._executeTriggers(triggers));
         return true;
     }
+  
+    _onPreUpdateToken(scene, id, update) {
+        if (!scene.isView) return true;
+        if (update.x === undefined && update.y === undefined) return true;
+        const token = scene.data.tokens.find(t => t._id === update._id);
+        if (token.hidden) return true; // don't stop ivnisible tokens?
+
+        // Get all trigger tokens in scene
+        let targets = this._getTriggerTokens(canvas.tokens.placeables, this.triggers, 'capture');
+        if (!targets) return true;
+
+        let x = update.x || token.x;
+        let  y = update.y || token.y;
+        let tw = token.width * canvas.scene.data.grid / 2;
+        let th = token.height * canvas.scene.data.grid / 2;
+        let motion = new Ray({x: token.x + tw, y: token.y + th}, {x: x + tw, y: y + th});
+
+        // don't trigger on tokens that are already captured
+        targets = targets.filter(target => token.x !== target.x + target.w  / 2 - tw || token.y !== target.y + target.h / 2 - th)
+        
+        // sort list by distance from start token position
+        targets.sort((a , b) => dist(token.x, token.y, b.x, b.y) - dist(token.x, token.y, a.x, a.y));
+        
+        targets.forEach(target => {
+            // test motion vs token diagonals
+            if (motion.intersectSegment([target.x, target.y, target.x + target.w, target.y + target.h])
+            || motion.intersectSegment([target.x, target.y + target.h, target.x + target.w, target.y])) {
+                update.x = target.x + target.w / 2 - tw;
+                update.y = target.y + target.h / 2 - th;
+                return true;
+            }
+        })
+        return true;
+    };
+}
+
+let dist = (x0, y0, x1, y1) => {

```

Comment:
I don't like a function like this being declared outside the scope (contaminating/conflicting with foundry.js or other modules), and that it's declared with `let name = arrowFunction`.
It's also not really needed, see comment above.

Reply:
Was not paying attention - I has assumed it was a module.